Sources/PrimerSDK/Classes/Core/Primer/Primer.swift

```diff
@@ -152,19 +155,28 @@ public class Primer {
      1.4.0
      */
     public func showCheckout(_ controller: UIViewController, flow: PrimerSessionFlow) {
+        self.presentingViewController = controller
+        Primer.shared.flow = flow
+        
         DispatchQueue.main.async { [weak self] in
-            self?.root = RootViewController()
-            guard let root = self?.root else { return }
-            let router: RouterDelegate = DependencyContainer.resolve()
-            let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
-            
-            if flow.internalSessionFlow.vaulted {
-                (settings as! PrimerSettings).amount = nil
+            if case .checkoutWithApplePay = flow {
+                let appleViewModel: ApplePayViewModelProtocol = DependencyContainer.resolve()
+                appleViewModel.payWithApple { (err) in
+                    
```

Comment:
Nope, cause the error management is performed (and exposed) in the view model. The major difference between Apple Pay and other services, is that Apple Pay doesn't present the root view controller (as the other services do). So on the others, the root view controller is acting as a proxy. Also, because the root view controller is responsible for showing success/error screen, it was made responsible to expose the errors. I don't agree with this, but it's something that will take quite some effort to be refactored, and I consider it out of scope of this ticket.